src/app/pages/orders-page/orders-page.component.ts

```diff
@@ -1,8 +1,241 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { combineLatest, map, Observable, Subject, switchMap, take, takeUntil } from 'rxjs';
+import { formatDate } from '@angular/common';
+import {
+  selectCarriages,
+  selectOrders,
+  selectStations,
+  selectUsers,
+} from '../../core/store/trips/trips.selectors';
+import {
+  deleteOrder,
+  loadCarriages,
+  loadOrders,
+  loadStations,
+  loadUsers,
+} from '../../core/store/trips/trips.actions';
+import { Order } from '../../features/trips/models/order.model';
+import { selectAccess } from '../../core/store/user/user.selectors';
+import { Access } from '../../core/models/user.model';
+import { Carriage } from '../../features/trips/models/carriage.model';
+import { Station } from '../../features/trips/models/station.model';
+import { OrderForView } from './order/order.component';
+import { User } from '../../features/trips/models/user.model';
+
+interface CarriageData {
+  number: number;
+  type: string;
+  seat: number;
+}
 
 @Component({
   selector: 'app-orders-page',
   templateUrl: './orders-page.component.html',
-  styleUrl: './orders-page.component.scss',
+  styleUrls: ['./orders-page.component.scss'],
 })
-export class OrdersPageComponent {}
+export class OrdersPageComponent implements OnInit, OnDestroy {
+  orders$: Observable<Order[] | null> = this.store.select(selectOrders);
+
+  stations$: Observable<Station[] | null> = this.store.select(selectStations);
+
+  carriages$: Observable<Carriage[] | null> = this.store.select(selectCarriages);
+
+  users$: Observable<User[] | null> = this.store.select(selectUsers);
+
+  role$!: Observable<Access>;
+
+  ordersForView!: OrderForView[];
+
+  warningOrderId: number | null = null;
+
+  private destroy$ = new Subject<void>();
+
+  constructor(private store: Store) {}
+
+  ngOnInit(): void {
+    /**/
+    // ________________
+
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    // this.store.dispatch(createOrder({ rideId: 2, seat: 91, stationStart: 41, stationEnd: 4 }));
+
+    // this.store.dispatch(createOrder({ rideId: 5, seat: 108, stationStart: 79, stationEnd: 71 }));
+
+    // ________________
+
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    this.store.dispatch(loadStations());
+
+    this.role$ = this.store.select(selectAccess);
+
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    this.store.dispatch(loadCarriages());
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    this.store.dispatch(loadUsers());
```

Comment:
Remove those comments please

Reply:
I removed the test order comment, but I can't do it without it:

// eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially

Reply:
maybe effect is needed here?

Reply:
You're right, I added a new action and a redusser effect for it.